protocol/flatbuffers/data_feed.fbs

```diff
@@ -0,0 +1,124 @@
+/// with a unified API.
+///
+/// The core datatype is `DeviceStatus`, which contains a table of optional fields,
+/// representing all of the data about a given tracked device. There is a corresponding
+/// mask on those fields called `DeviceStatusMask`, which defaults to all fields
+/// false, meaning no data.
+///
+/// It is important to note that the absence of a value in a serialized flatbuffer
+/// binary corresponds to the default value in the schema, which means that only
+/// fields that are actually populated have any impact on the bandwidth of data
+/// used by flatbuffer messages. This allows us to lump basically all fields that
+/// could possibly be needed into `DeviceStatus`, without impacting performance.
+///
+/// There are two ways of getting `DeviceStatus`es - one way is by requesting a single
+/// update, via `PollDeviceStatus`. Another way is by requesting a `DataFeed` via a
+/// `DataFeedRequest`, which itself is simply a list of `DataFeedConfig`s, one for each
+/// data feed that is being requested.
+///
+/// `DataFeedConfig` allows fine grained control over the type of data sent (via
+/// `DeviceStatusMask`), and the frequency of that data.
+
+
+include "datatypes/hardware_info.fbs";
+include "datatypes/datatypes.fbs";
+include "datatypes/imu.fbs";
+
+namespace slimevr_protocol.data_feed;
+
+/// Requests for a single `DeviceStatus` to be sent. This is helpful when getting
+/// initial info about the device.
+table PollDeviceStatus {
+    mask: DeviceStatusMask;
+}
+
+
+/// Requests for the other party to send `data_feeds`.
+/// For example, GUI requests for position data to be sent from server.
+//
+// TODO: Should we add a `DataFeedCancel`? I think its fine to just kill the connection
+//   to do that. We don't need *that* much control.
+table DataFeedRequest {
+    data_feeds: [DataFeedConfig];
+}
+
+/// Notifies the other party that `data_feeds` will be sent.
+/// For example, feeder app notifies server that controller data will be sent.
+//
+// TODO: Perhaps its enough to just start sending the data feed info? Do we really
+//   need to notify the other end that we are going to do it? Or perhaps its also useful
+//   to respond to a `DataFeedRequest` and tell them which feeds will *actually* be
+//   supported. I'm not sold on the necessity of this either way, I'd rather just
+//   let the application handle the case of data being missing, and respond to incoming
+//   data feeds automatically.
+table DataFeedNotify {
+    data_feeds: [DataFeedConfig];
+}
+
+
+/// All information related to the configuration of a data feed. This may be sent
+/// as part of a `DataFeedRequest` or a `DataFeedNotify`.
+table DataFeedConfig {
+    /// Minimum delay in milliseconds between new data updates.
+    min_interval: uint16;
+    data_mask: DeviceStatusMask;
+
+    // Note: I removed control over only sending on update, etc, because it seemed
+    // unecessarily complicated to implement
```

Comment:
Why do you think it's hard to implement a event based broadcasting system? I feel like the interval based solution will be harder to implement.

Reply:
the event based solution make de protocol possibly send some packets all the time, and this can be not practical in the case of an UI.
Lets say you send a packet based on a change in the position / rotation of the tracker. That mean the ui will receive realtime (like 1000Hz) packet for position upate. But the ui does not need that kind of speed.

Having a minimum interval request make it so you can say how "offten" you want each kind of data.

The ui could ask for a 100Hz refresh rate for
But the overlay can ask for real time data.

it is also easier to implement in the server in its current state imo. Because we have the tick function that can give us a snapshot of all the datas at that current time. instead of having to make linsteners for everyting to have an event based implementation

Reply:
Perhaps it might be worth separating "data we likely want as events" and "data we likely want as a stream?" like, we never need a stream of name updates, even 1/10 hz or something like that could be considered too often and bandwidth-wasting.

Reply:
this ^

Reply:
Resolved via the RPC system